Fix pandas 3.* compat

```
FAILED dask_geomodeling/tests/test_geometry.py::TestWhere::test_mask_with_float_filter - AssertionError: False is not true
FAILED dask_geomodeling/tests/test_geometry.py::TestFieldOperations::test_classify_not_categorical - AssertionError: 'str' != 'object'
- str
+ object
FAILED dask_geomodeling/tests/test_geometry.py::TestText::test_parser_into_same_column - AssertionError: 'object' != 'str'
- object
+ str
FAILED dask_geomodeling/tests/test_geometry.py::TestText::test_parser_two_same - AssertionError: 'object' != 'str'
- object
+ str
FAILED dask_geomodeling/tests/test_raster_temporal.py::test_period[M-left-left-UTC-expected13] - ValueError: Invalid frequency: M. Failed to parse with error message: ValueError("'M' is no longer supported for offsets. Please use 'ME' instead.")
FAILED dask_geomodeling/tests/test_raster_temporal.py::test_period[M-left-right-UTC-expected14] - ValueError: Invalid frequency: M. Failed to parse with error message: ValueError("'M' is no longer supported for offsets. Please use 'ME' instead.")
FAILED dask_geomodeling/tests/test_raster_temporal.py::test_period[M-right-left-UTC-expected15] - ValueError: Invalid frequency: M. Failed to parse with error message: ValueError("'M' is no longer supported for offsets. Please use 'ME' instead.")
FAILED dask_geomodeling/tests/test_raster_temporal.py::test_period[M-right-right-UTC-expected16] - ValueError: Invalid frequency: M. Failed to parse with error message: ValueError("'M' is no longer supported for offsets. Please use 'ME' instead.")
FAILED dask_geomodeling/tests/test_raster_temporal.py::test_period[M-None-None-UTC-expected17] - ValueError: Invalid frequency: M. Failed to parse with error message: ValueError("'M' is no longer supported for offsets. Please use 'ME' instead.")
FAILED dask_geomodeling/tests/test_raster_temporal.py::test_period[H-left-left-UTC-expected27] - ValueError: Invalid frequency: H. Failed to parse with error message: ValueError("Invalid frequency: H. Failed to parse with error message: KeyError('H'). Did you mean h?") Did you mean h?
FAILED dask_geomodeling/tests/test_raster_temporal.py::test_period[M-left-left-UTC-expected28] - ValueError: Invalid frequency: M. Failed to parse with error message: ValueError("'M' is no longer supported for offsets. Please use 'ME' instead.")
FAILED dask_geomodeling/tests/test_raster_temporal.py::TestTemporalAggregate::test_count_dtype - ValueError: Invalid frequency: M. Failed to parse with error message: ValueError("'M' is no longer supported for offsets. Please use 'ME' instead.")
FAILED dask_geomodeling/tests/test_raster_temporal.py::TestTemporalAggregate::test_get_data_count - ValueError: Invalid frequency: M. Failed to parse with error message: ValueError("'M' is no longer supported for offsets. Please use 'ME' instead.")
FAILED dask_geomodeling/tests/test_raster_temporal.py::TestTemporalAggregate::test_get_data_max - ValueError: Invalid frequency: M. Failed to parse with error message: ValueError("'M' is no longer supported for offsets. Please use 'ME' instead.")
FAILED dask_geomodeling/tests/test_raster_temporal.py::TestTemporalAggregate::test_get_data_mean - ValueError: Invalid frequency: M. Failed to parse with error message: ValueError("'M' is no longer supported for offsets. Please use 'ME' instead.")
FAILED dask_geomodeling/tests/test_raster_temporal.py::TestTemporalAggregate::test_get_data_median - ValueError: Invalid frequency: M. Failed to parse with error message: ValueError("'M' is no longer supported for offsets. Please use 'ME' instead.")
FAILED dask_geomodeling/tests/test_raster_temporal.py::TestTemporalAggregate::test_get_data_min - ValueError: Invalid frequency: M. Failed to parse with error message: ValueError("'M' is no longer supported for offsets. Please use 'ME' instead.")
FAILED dask_geomodeling/tests/test_raster_temporal.py::TestTemporalAggregate::test_get_data_percentile - ValueError: Invalid frequency: M. Failed to parse with error message: ValueError("'M' is no longer supported for offsets. Please use 'ME' instead.")
FAILED dask_geomodeling/tests/test_raster_temporal.py::TestTemporalAggregate::test_get_data_std - ValueError: Invalid frequency: M. Failed to parse with error message: ValueError("'M' is no longer supported for offsets. Please use 'ME' instead.")
FAILED dask_geomodeling/tests/test_raster_temporal.py::TestTemporalAggregate::test_get_data_sum - ValueError: Invalid frequency: M. Failed to parse with error message: ValueError("'M' is no longer supported for offsets. Please use 'ME' instead.")
FAILED dask_geomodeling/tests/test_raster_temporal.py::TestTemporalAggregate::test_get_data_sum_hour - ValueError: Invalid frequency: H. Failed to parse with error message: ValueError("Invalid frequency: H. Failed to parse with error message: KeyError('H'). Did you mean h?") Did you mean h?
FAILED dask_geomodeling/tests/test_raster_temporal.py::TestTemporalAggregate::test_get_data_time_request - ValueError: Invalid frequency: H. Failed to parse with error message: ValueError("Invalid frequency: H. Failed to parse with error message: KeyError('H'). Did you mean h?") Did you mean h?
FAILED dask_geomodeling/tests/test_raster_temporal.py::TestTemporalAggregate::test_get_data_var - ValueError: Invalid frequency: M. Failed to parse with error message: ValueError("'M' is no longer supported for offsets. Please use 'ME' instead.")
FAILED dask_geomodeling/tests/test_raster_temporal.py::TestTemporalAggregate::test_int_result_dtype - ValueError: Invalid frequency: M. Failed to parse with error message: ValueError("'M' is no longer supported for offsets. Please use 'ME' instead.")
FAILED dask_geomodeling/tests/test_raster_temporal.py::TestTemporalAggregate::test_min_max_dtype - ValueError: Invalid frequency: M. Failed to parse with error message: ValueError("'M' is no longer supported for offsets. Please use 'ME' instead.")
FAILED dask_geomodeling/tests/test_raster_temporal.py::TestTemporalAggregate::test_other_dtype - ValueError: Invalid frequency: M. Failed to parse with error message: ValueError("'M' is no longer supported for offsets. Please use 'ME' instead.")
FAILED dask_geomodeling/tests/test_raster_temporal.py::TestTemporalAggregate::test_sum_dtype - ValueError: Invalid frequency: M. Failed to parse with error message: ValueError("'M' is no longer supported for offsets. Please use 'ME' instead.")
FAILED dask_geomodeling/tests/test_raster_temporal.py::TestTemporalAggregate::test_temporal - ValueError: Invalid frequency: M. Failed to parse with error message: ValueError("'M' is no longer supported for offsets. Please use 'ME' instead.")
FAILED dask_geomodeling/tests/test_raster_temporal.py::TestTemporalAggregate::test_timedelta - ValueError: Invalid frequency: S. Failed to parse with error message: ValueError("Invalid frequency: S. Failed to parse with error message: KeyError('S'). Did you mean s?") Did you mean s?
FAILED dask_geomodeling/tests/test_raster_temporal.py::TestCumulative::test_get_data_count - ValueError: Invalid frequency: M. Failed to parse with error message: ValueError("'M' is no longer supported for offsets. Please use 'ME' instead.")
FAILED dask_geomodeling/tests/test_raster_temporal.py::TestCumulative::test_get_data_sum_month - ValueError: Invalid frequency: M. Failed to parse with error message: ValueError("'M' is no longer supported for offsets. Please use 'ME' instead.")
FAILED dask_geomodeling/tests/test_raster_temporal.py::test_resample_period[H-backward-Atlantic/Azores-expected9] - ValueError: Invalid frequency: H. Failed to parse with error message: ValueError("Invalid frequency: H. Failed to parse with error message: KeyError('H'). Did you mean h?") Did you mean h?
FAILED dask_geomodeling/tests/test_raster_temporal.py::test_resample_timedelta[H-expected0] - ValueError: Invalid frequency: H. Failed to parse with error message: ValueError("Invalid frequency: H. Failed to parse with error message: KeyError('H'). Did you mean h?") Did you mean h?
FAILED dask_geomodeling/tests/test_raster_temporal.py::test_resample_timedelta[D-expected1] - AssertionError: assert None == datetime.timedelta(days=1)
 +  where None = Resample(MemorySource(array([[[  1,   1,   1,   1,   1,   1,   1,   1,   1,   1],\n        [  1,   1,   1,   1,   1,   ..., 946684800000, 3600000, ['Testmeta for band 0', 'Testmeta for band 1', 'Testmeta for band 2']), 'D', 'nearest', 'UTC').timedelta
FAILED dask_geomodeling/tests/test_raster_temporal.py::test_resample_timedelta[15T-expected2] - ValueError: Invalid frequency: 15T. Failed to parse with error message: ValueError("Invalid frequency: T. Failed to parse with error message: KeyError('T'). Did you mean min?")
FAILED dask_geomodeling/tests/test_raster_temporal.py::test_resample_timedelta[S-expected3] - ValueError: Invalid frequency: S. Failed to parse with error message: ValueError("Invalid frequency: S. Failed to parse with error message: KeyError('S'). Did you mean s?") Did you mean s?
FAILED dask_geomodeling/tests/test_raster_temporal.py::test_resample_timedelta[M-None] - ValueError: Invalid frequency: M. Failed to parse with error message: ValueError("'M' is no longer supported for offsets. Please use 'ME' instead.")
FAILED dask_geomodeling/tests/test_raster_temporal.py::test_resample_timedelta[A-None] - ValueError: Invalid frequency: A. Failed to parse with error message: ValueError("Invalid frequency: A. Failed to parse with error message: KeyError('A'). Did you mean Y?") Did you mean Y?
```

--- a/.github/workflows/test-conda.yaml
+++ b/.github/workflows/test-conda.yaml
@@ -32,7 +32,7 @@ jobs:
 
       - name: Setup Environment
         run: |
-          conda create --quiet --name test python=${{ matrix.python }} pytest numpy gdal scipy pytz dask-core toolz pandas==2.* geopandas pyproj pyogrio
+          conda create --quiet --name test python=${{ matrix.python }} pytest numpy gdal scipy pytz dask-core toolz pandas geopandas pyproj pyogrio
           python -V
           conda info
           conda list

--- a/CHANGES.rst
+++ b/CHANGES.rst
@@ -10,6 +10,7 @@ Changelog of dask-geomodeling
 - The GeometryBlock.to_file now automatically determines the 'geometry' to write if
   not given.
 
+- Pandas 3.* compatibility fixes.
 
 2.7.0 (2026-01-19)
 ------------------

--- a/dask_geomodeling/raster/temporal.py
+++ b/dask_geomodeling/raster/temporal.py
@@ -17,6 +17,7 @@ from dask_geomodeling.utils import (
     dtype_for_statistic,
     find_neigbours,
     normalize_offset,
+    offset_to_timedelta,
 )
 
 from .base import RasterBlock, BaseSingle
@@ -604,10 +605,7 @@ class TemporalAggregate(BaseSingle):
     def timedelta(self):
         if self.frequency is None:
             return None
-        try:
-            return pd.Timedelta(to_offset(self.frequency)).to_pytimedelta()
-        except ValueError:
-            return  # e.g. Month is non-equidistant
+        return offset_to_timedelta(self.frequency)
 
     @property
     def temporal(self):
@@ -1128,10 +1126,7 @@ class Resample(BaseSingle):
 
     @property
     def timedelta(self):
-        try:
-            return pd.Timedelta(to_offset(self.frequency)).to_pytimedelta()
-        except ValueError:
-            return  # e.g. Month is non-equidistant
+        return offset_to_timedelta(self.frequency)
 
     def get_sources_and_requests(self, **request):
         process_kwargs = {

--- a/dask_geomodeling/tests/test_geometry.py
+++ b/dask_geomodeling/tests/test_geometry.py
@@ -8,7 +8,7 @@ import geopandas as gpd
 import numpy as np
 import pandas as pd
 
-from dask_geomodeling.utils import Extent, get_sr, shapely_transform
+from dask_geomodeling.utils import Extent, IS_PANDAS_SINCE_3, get_sr, shapely_transform
 from dask_geomodeling.tests.factories import (
     setup_temp_root,
     teardown_temp_root,
@@ -841,7 +841,7 @@ class TestWhere(unittest.TestCase):
         view = self.source.set("result", series)
 
         result = view.get_data(**self.request)
-        expected = pd.Series(["Hola!", "Hola!", "Hola!", "Hola!", "Hola!", float("nan")])
+        expected = pd.Series(["Hola!", "Hola!", "Hola!", "Hola!", "Hola!", float("nan")], dtype=object)
         self.assertTrue(result["features"]["result"].equals(expected))
 
     def test_mask_with_other_column(self):
@@ -1133,7 +1133,8 @@ class TestFieldOperations(unittest.TestCase):
         actual = field_operations.Classify(
             self.source["col_source"], bins=[0, 0.5, 1.0], labels=["A", "B", "C", "D"]
         ).get_data(**self.request)
-        self.assertEqual(actual.dtype.name, "object")
+        expected_dtype = "str" if IS_PANDAS_SINCE_3 else "object"
+        self.assertEqual(actual.dtype.name, expected_dtype)
 
     def test_classify_from_columns_left(self):
         source_with_bins = self.source.set("bin_1", 0, "bin_2", 1.2, "bin_3", 5.0)
@@ -1573,7 +1574,8 @@ class TestText(unittest.TestCase):
         )
         view = text.ParseTextColumn(source, "description", self.key_mapping)
         data = view.get_data(**self.request)["features"]
-        self.assertEqual("object", str(data["model_name"].dtype))
+        expected_dtype = "str" if IS_PANDAS_SINCE_3 else "object"
+        self.assertEqual(str(data["model_name"].dtype), expected_dtype)
         for col in self.expected:
             assert data.loc[1, col] == self.expected[col]
             assert data.loc[2, col] == self.expected[col]
@@ -1627,7 +1629,8 @@ class TestText(unittest.TestCase):
             self.source, "description", {"modelname": "description"}
         )
         data = view.get_data(**self.request)["features"]
-        self.assertEqual("object", str(data["description"].dtype))
+        expected_dtype = "str" if IS_PANDAS_SINCE_3 else "object"
+        self.assertEqual(str(data["description"].dtype), expected_dtype)
         assert data.loc[1, "description"] == "rotterdam 01"
 
     def test_parser_into_same_column_non_existing(self):

--- a/dask_geomodeling/tests/test_geometry_sinks.py
+++ b/dask_geomodeling/tests/test_geometry_sinks.py
@@ -13,12 +13,29 @@ from dask_geomodeling.tests.factories import (
     setup_temp_root,
     teardown_temp_root,
 )
+from dask_geomodeling.utils import IS_PANDAS_SINCE_3
+
+
+def _coerce_string_cols_to_object(gdf):
+    """Coerce pandas 3 'str' dtype columns to 'object' dtype.
+
+    geopandas' read_file still returns 'object' dtype for string columns,
+    while pandas 3 infers 'str' dtype for in-memory string data. Coercing
+    to object keeps both sides comparable.
+    """
+    for col in gdf.columns:
+        if gdf[col].dtype.name == "str":
+            gdf[col] = gdf[col].astype(object)
+    return gdf
 
 
 def assert_frame_equal_ignore_index(actual, expected, sort_col, precision=None):
     if precision:
         actual.geometry = actual.geometry.set_precision(precision)
         expected.geometry = expected.geometry.set_precision(precision)
+    if IS_PANDAS_SINCE_3:
+        actual = _coerce_string_cols_to_object(actual)
+        expected = _coerce_string_cols_to_object(expected)
     assert_geodataframe_equal(
         actual.set_index(sort_col).sort_index(),
         expected.set_index(sort_col).sort_index(),

--- a/dask_geomodeling/tests/test_raster_temporal.py
+++ b/dask_geomodeling/tests/test_raster_temporal.py
@@ -53,12 +53,12 @@ td = Timedelta
     ("h", "left", "left", "Europe/Amsterdam", (dt(2000, 1, 1), dt(2000, 1, 3))),
     ("h", "left", "left", "America/New_York", (dt(2000, 1, 1), dt(2000, 1, 3))),
     (None, "left", "left", "UTC", (dt(2000, 1, 3), dt(2000, 1, 3))),
-    # pandas MonthEnd ("M") bin is 1999-12-31 00:00 UTC to 2000-01-31 00:00 UTC
-    ("M", "left", "left", "UTC", (dt(1999, 12, 31), dt(1999, 12, 31))),
-    ("M", "left", "right", "UTC", (dt(2000, 1, 31), dt(2000, 1, 31))),
-    ("M", "right", "left", "UTC", (dt(1999, 12, 31), dt(1999, 12, 31))),
-    ("M", "right", "right", "UTC", (dt(2000, 1, 31), dt(2000, 1, 31))),
-    ("M", None, None, "UTC", (dt(2000, 1, 31), dt(2000, 1, 31))),  # (right, right)
+    # pandas MonthEnd ("ME") bin is 1999-12-31 00:00 UTC to 2000-01-31 00:00 UTC
+    ("ME", "left", "left", "UTC", (dt(1999, 12, 31), dt(1999, 12, 31))),
+    ("ME", "left", "right", "UTC", (dt(2000, 1, 31), dt(2000, 1, 31))),
+    ("ME", "right", "left", "UTC", (dt(1999, 12, 31), dt(1999, 12, 31))),
+    ("ME", "right", "right", "UTC", (dt(2000, 1, 31), dt(2000, 1, 31))),
+    ("ME", None, None, "UTC", (dt(2000, 1, 31), dt(2000, 1, 31))),  # (right, right)
     # pandas MonthStart ("MS") bin is 2000-01-01 00:00 UTC to 2000-02-01 00:00 UTC
     ("MS", "left", "left", "UTC", (dt(2000, 1, 1), dt(2000, 1, 1))),
     ("MS", "left", "right", "UTC", (dt(2000, 2, 1), dt(2000, 2, 1))),
@@ -71,7 +71,7 @@ td = Timedelta
     # closed=right moves "monday" into the weekend because it is on 00:00
     ("B", "right", "left", "UTC", (dt(1999, 12, 31), dt(1999, 12, 31))),
     ("B", "right", "right", "UTC", (dt(2000, 1, 3), dt(2000, 1, 3))),
-    # Aliases deprecated since pandas 2.2 still work
+    # Aliases deprecated since pandas 2.2 still work (normalized by normalize_offset)
     ("H", "left", "left", "UTC", (dt(2000, 1, 1, 0), dt(2000, 1, 3, 0))),
     ("M", "left", "left", "UTC", (dt(1999, 12, 31), dt(1999, 12, 31))),
 ])
@@ -614,8 +614,10 @@ class TestCumulative(unittest.TestCase):
         # Source period in Azores (GMT -1) timezone is [1999-12-31 23:00, 2000-01-01 01:00]
         # this affects the binning for daily frequency. With backwards direction, the period
         # will be [2000-01-01, 2000-01-02] Azores midnight. We add one hour to convert to UTC.
-        ("H", "backward", "Atlantic/Azores", (dt(2000, 1, 1), dt(2000, 1, 1, 2))),
+        ("h", "backward", "Atlantic/Azores", (dt(2000, 1, 1), dt(2000, 1, 1, 2))),
         ("D", "backward", "Atlantic/Azores", (dt(2000, 1, 1, 1), dt(2000, 1, 2, 1))),
+        # Deprecated alias still works (translated by normalize_offset)
+        ("H", "backward", "UTC", (dt(2000, 1, 1), dt(2000, 1, 1, 2))),
     ],
 )
 def test_resample_period(freq, direction, timezone, expected, source):
@@ -631,11 +633,16 @@ def test_resample_period_empty(empty_source):
 @pytest.mark.parametrize(
     "freq,expected,",
     [
-        ("H", Timedelta(hours=1)),
+        ("h", Timedelta(hours=1)),
         ("D", Timedelta(days=1)),
-        ("15T", Timedelta(minutes=15)),
-        ("S", Timedelta(seconds=1)),
+        ("15min", Timedelta(minutes=15)),
+        ("s", Timedelta(seconds=1)),
         ("MS", None),
+        ("ME", None),
+        ("YE", None),
+        # Deprecated aliases still work (translated by normalize_offset)
+        ("H", Timedelta(hours=1)),
+        ("S", Timedelta(seconds=1)),
         ("M", None),
         ("A", None),
     ],

--- a/dask_geomodeling/utils.py
+++ b/dask_geomodeling/utils.py
@@ -1018,8 +1018,6 @@ def _translate_removed_freq_aliases(freq: str) -> str:
 
     Handles both simple aliases like 'M' -> 'ME' and prefixed ones like '2M' -> '2ME'.
     """
-    import re
-    # Match optional numeric prefix + alias
     match = re.match(r'^(\d*)(.+)$', freq)
     if match:
         prefix, alias = match.groups()

--- a/dask_geomodeling/utils.py
+++ b/dask_geomodeling/utils.py
@@ -33,9 +33,30 @@ POLYGON = "POLYGON (({0} {1},{2} {1},{2} {3},{0} {3},{0} {1}))"
 GDAL3 = gdal.VersionInfo().startswith("3")
 try:
     IS_PANDAS_SINCE_2_2 = Version(pd.__version__) >= Version("2.2.0")
+    IS_PANDAS_SINCE_3 = Version(pd.__version__) >= Version("3.0.0")
 except TypeError:  # for doc-build, version is mocked
     IS_PANDAS_SINCE_2_2 = True
+    IS_PANDAS_SINCE_3 = True
 PANDAS_2_2_NEW_FREQS = {"ME", "BME", "SME", "CBME", "QE", "BQE", "YE", "BYE"}
+# Mapping of frequency aliases removed in pandas 3.0.
+# All entries below come from the pandas 3.0.0 release notes:
+# https://pandas.pydata.org/docs/whatsnew/v3.0.0.html#removal-of-prior-version-deprecations-changes
+#
+# GH#57986 – end-of-period offsets renamed (M→ME, BM→BME, etc.)
+# GH#57699, GH#57793 – annual aliases (A→YE, BA→BYE, AS→YS, BAS→BYS)
+# GH#59143 – hour aliases (H→h, BH→bh, CBH→cbh)
+# GH#57627 – sub-hour aliases (T→min, S→s, L→ms, U→us, N→ns)
+PANDAS_3_REMOVED_ALIASES = {
+    # End-of-period offsets (GH#57986)
+    "M": "ME", "BM": "BME", "SM": "SME", "CBM": "CBME",
+    "Q": "QE", "BQ": "BQE", "Y": "YE", "BY": "BYE",
+    # Annual aliases (GH#57699, GH#57793)
+    "A": "YE", "BA": "BYE", "AS": "YS", "BAS": "BYS",
+    # Hour aliases (GH#59143)
+    "H": "h", "BH": "bh", "CBH": "cbh",
+    # Sub-hour aliases (GH#57627)
+    "T": "min", "S": "s", "L": "ms", "U": "us", "N": "ns",
+}
 
 def get_index(values, no_data_value):
     """ Return an index to access for data values in values. """
@@ -936,6 +957,25 @@ def find_neigbours(array, value, direction="nearest"):
     
     return np.clip(indices, 0, array.size - 1)
 
+def offset_to_timedelta(freq):
+    """Convert a pandas frequency string to a python timedelta, or None.
+
+    Returns None for non-equidistant frequencies like MonthEnd.
+
+    Args:
+        freq (str): pandas frequency string
+
+    Returns:
+        datetime.timedelta or None
+    """
+    try:
+        offset = to_offset(freq)
+        td = pd.Timedelta(offset.nanos, unit="ns")
+        return td.to_pytimedelta()
+    except (ValueError, AttributeError):
+        return None  # e.g. MonthEnd is non-equidistant
+
+
 def normalize_offset(freq) -> str:
     """Convert pandas frequency strings to compatible with the current pandas.
 
@@ -953,12 +993,36 @@ def normalize_offset(freq) -> str:
     """
     if freq is None:
         return None
-    if not IS_PANDAS_SINCE_2_2 and any(freq.endswith(x) for x in PANDAS_2_2_NEW_FREQS):
-        return freq[:-1]  # chop off the 'E'
-    elif not IS_PANDAS_SINCE_2_2:
-        offset = to_offset(freq)
-    else:
-        with warnings.catch_warnings():
-            warnings.simplefilter("ignore", FutureWarning)
-            offset = to_offset(freq)
-    return offset.freqstr
+
+    # pandas < 2.2: new-style "ME"/"YE"/etc. suffixes aren't recognized yet,
+    # so strip the trailing "E" to get the old-style alias.
+    if not IS_PANDAS_SINCE_2_2:
+        if any(freq.endswith(x) for x in PANDAS_2_2_NEW_FREQS):
+            return freq[:-1]
+        return to_offset(freq).freqstr
+
+    # pandas >= 3.0: old-style aliases ("M", "H", "S", etc.) have been removed,
+    # so translate them to the new form before calling to_offset.
+    if IS_PANDAS_SINCE_3:
+        freq = _translate_removed_freq_aliases(freq)
+        return to_offset(freq).freqstr
+
+    # pandas 2.2.x: old-style aliases still work but emit FutureWarning.
+    with warnings.catch_warnings():
+        warnings.simplefilter("ignore", FutureWarning)
+        return to_offset(freq).freqstr
+
+
+def _translate_removed_freq_aliases(freq: str) -> str:
+    """Translate frequency aliases removed in pandas 3.0 to their replacements.
+
+    Handles both simple aliases like 'M' -> 'ME' and prefixed ones like '2M' -> '2ME'.
+    """
+    import re
+    # Match optional numeric prefix + alias
+    match = re.match(r'^(\d*)(.+)$', freq)
+    if match:
+        prefix, alias = match.groups()
+        if alias in PANDAS_3_REMOVED_ALIASES:
+            return prefix + PANDAS_3_REMOVED_ALIASES[alias]
+    return freq

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -22,7 +22,7 @@ classifiers = [
 ]
 dependencies = [
     "dask[delayed]>=2022",
-    "pandas>=2,<3",
+    "pandas>=2",
     "geopandas>=1",
     "pytz",
     "numpy>=1.23",

